modules/edu.gemini.seqexec.server/src/test/scala/edu/gemini/seqexec/server/SeqexecEngineSpec.scala

```diff
@@ -155,128 +156,184 @@ class SeqexecEngineSpec extends FlatSpec with Matchers {
         seqId2 -> sequence(seqId2),
         seqId3 -> sequence(seqId3)
       ) )
 
-    def testAdvance(obsId: SPObservationID, n: Int): Option[EngineState] = {
-      val q = async.boundedQueue[EngineEvent](10)
-      advanceOne(q, s0, seqexecEngine.moveSequenceInQueue(q, queueName, obsId, n))
+    def testAdvance(obsId: SPObservationID, n: Int): Option[executeEngine.StateType] = {
+      val q = async.boundedQueue[executeEngine.EventType](10)
+      advanceOne(q, s0, seqexecEngine.moveSequenceInQueue(q, CalibrationQueueName, obsId, n))
     }
 
     val sf1 = testAdvance(seqObsId2, -1)
 
-    inside(sf1.flatMap(_.userData.get(queueName))) {
+    inside(sf1.flatMap(x => queuesL.get(x.userData).get(CalibrationQueueName))) {
       case Some(exq) => exq shouldBe List(seqId2, seqId1, seqId3)
     }
 
     val sf2 = testAdvance(seqObsId1, 2)
 
-    inside(sf2.flatMap(_.userData.get(queueName))) {
+    inside(sf2.flatMap(x => queuesL.get(x.userData).get(CalibrationQueueName))) {
       case Some(exq) => exq shouldBe List(seqId2, seqId3, seqId1)
     }
 
     val sf3 = testAdvance(seqObsId3, 4)
 
-    inside(sf3.flatMap(_.userData.get(queueName))) {
+    inside(sf3.flatMap(x => queuesL.get(x.userData).get(CalibrationQueueName))) {
       case Some(exq) => exq shouldBe List(seqId1, seqId2, seqId3)
     }
 
     val sf4 = testAdvance(seqObsId1, -2)
 
-    inside(sf4.flatMap(_.userData.get(queueName))) {
+    inside(sf4.flatMap(x => queuesL.get(x.userData).get(CalibrationQueueName))) {
       case Some(exq) => exq shouldBe List(seqId1, seqId2, seqId3)
     }
   }
 
+  "SeqexecEngine setOperator" should "set operator's name" in {
+    val operator = Operator("Joe")
+    val s0 = Engine.State.empty(EngineMetadata.default)
+    val q = async.boundedQueue[executeEngine.EventType](10)
+    val sf = advanceN(q, s0, seqexecEngine.setOperator(q, UserDetails("", ""), operator), 2)
+
+    inside(sf.flatMap((Engine.State.userDataL ^|-> EngineMetadata.operatorL).get(_))) {
+      case Some(op) => op shouldBe operator
+    }
+
+  }
+
+  "SeqexecEngine setImageQuality" should "set Image Quality condition" in {
+    val iq = ImageQuality.Percent20
+    val s0 = Engine.State.empty(EngineMetadata.default)
+    val q = async.boundedQueue[executeEngine.EventType](10)
+    val sf = advanceN(q, s0, seqexecEngine.setImageQuality(q, iq, UserDetails("", "")), 2)
+
+    inside(sf.map((Engine.State.userDataL ^|-> EngineMetadata.conditionsL ^|-> Conditions.iq).get(_))) {
+      case Some(op) => op shouldBe iq
+    }
+
+  }
+
+  "SeqexecEngine setWaterVapor" should "set Water Vapor condition" in {
+    val wv = WaterVapor.Percent80
+    val s0 = Engine.State.empty(EngineMetadata.default)
+    val q = async.boundedQueue[executeEngine.EventType](10)
+    val sf = advanceN(q, s0, seqexecEngine.setWaterVapor(q, wv, UserDetails("", "")), 2)
+
+    inside(sf.map((Engine.State.userDataL ^|-> EngineMetadata.conditionsL ^|-> Conditions.wv).get(_))) {
```

Comment:
inside?

Reply:
Yes, from scalatest. It is used all over the tests.